spm/private/spm_linux_toolchain.bzl

```diff
@@ -21,7 +21,8 @@ Could not parse the version number for Swift package manager. {version}\
     return current_version >= desired_version_value
 
 def _create_build_tool_config(ctx, target_triple, spm_configuration):
-    swift_worker = ctx.executable._swift_worker
+    swift_worker = ctx.attr._swift_worker[DefaultInfo].files_to_run.executable
```

Comment:
The `.executable` at the end makes this a no-op change. You want to pass the provider to `additional_tools`. You might need to use `.executable` in `args` though.

Reply:
You are correct. I have updated the code to use the `.executable` for the arg, but I pass along the `FilesToRunProvider` to `additional_tools` which, in turn, is passed to `execution_requirements` for `ctx.actions.run()`.

I have run integration tests specifically referencing [the commit that you suggested](https://github.com/bazelbuild/rules_swift/commit/8aca87eba7dc9164c9835b887ff58d1c8ddbf20e). I believe that the runfiles are being propagated properly.